src/Template.php

```diff
@@ -478,6 +478,35 @@ public function yieldParentBlock($name, array $context, array $blocks = []): ite
         }
     }
 
+    protected function hasMacro(string $name, array $context): bool
+    {
+        if (method_exists($this, $name)) {
+            return true;
+        }
+
+        if (!$parent = $this->getParent($context)) {
```

Comment:
The current code didn't have a look at parent templates for the "is defined" test.